addons/lunch/models/lunch_order.py

```diff
@@ -138,26 +138,21 @@ def create(self, vals_list):
         return orders
 
     def write(self, values):
-        merge_needed = 'note' in values or 'topping_ids_1' in values or 'topping_ids_2' in values or 'topping_ids_3' in values
+        merge_needed = 'note' in values or 'topping_ids_1' in values or 'topping_ids_2' in values or 'topping_ids_3' in values or 'state' in values
         default_location_id = self.env.user.last_lunch_location_id and self.env.user.last_lunch_location_id.id or False
 
         if merge_needed:
             lines_to_deactivate = self.env['lunch.order']
             for line in self:
-                # Only write on topping_ids_1 because they all share the same table
-                # and we don't want to remove all the records
-                # _extract_toppings will pop topping_ids_1, topping_ids_2 and topping_ids_3 from values
-                # This also forces us to invalidate the cache for topping_ids_2 and topping_ids_3 that
-                # could have changed through topping_ids_1 without the cache knowing about it
                 toppings = self._extract_toppings(values)
-                self.invalidate_model(['topping_ids_2', 'topping_ids_3'])
-                values['topping_ids_1'] = [(6, 0, toppings)]
```

Comment:
Are you sure about these lines because I'm not completly sure that the behavior still correct without it and doesn't seems to be in the specs

Reply:
you're right; my bad